src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/DSA.cs

```diff
@@ -85,15 +85,11 @@ public static DSA Create(DSAParameters parameters)
 
         public abstract bool VerifySignature(byte[] rgbHash, byte[] rgbSignature);
 
-        protected virtual byte[] HashData(byte[] data, int offset, int count, HashAlgorithmName hashAlgorithm)
-        {
-            throw DerivedClassMustOverride();
-        }
+        protected virtual byte[] HashData(byte[] data, int offset, int count, HashAlgorithmName hashAlgorithm) =>
+            HashOneShotHelpers.HashData(hashAlgorithm, new ReadOnlySpan<byte>(data, offset, count));
```

Comment:
Not sure if it's worth adding a filter to the DSA base class that blocks out MD5, since FIPS-186-{1,2} DSA is only SHA-1 and FIPS-186-{3,4} DSA is SHA-1 or SHA-2... but no DSA is MD5.  (and in FIPS-186-5-draft DSA has been completely deprecated, still no MD5 :smile:)

Reply:
@bartonjs For better or for worse, we allow MD5 today:

```C#
using System;
using System.Security.Cryptography;

DSAOpenSsl dsa = new DSAOpenSsl();
Console.WriteLine(Convert.ToHexString(dsa.SignData(new byte[1], 0, 1, HashAlgorithmName.MD5)));
```

I think in that case we'd end up breaking the snippet above.